### PR TITLE
add new API paddle.nn.initializer.Orthogonal and calculate_gain

### PR DESCRIPTION
**Pull request by zhwesky2010**
### PR types

New features
### PR changes

APIs
### Describe

paddle.nn.initializer.Orthogonal
---

infoflow 2021-11-18 21-39-01

```
weight_attr = paddle.ParamAttr(initializer=paddle.nn.initializer.Orthogonal())
linear = paddle.nn.Linear(10, 15, weight_attr=weight_attr)
# linear.weight: X * X' = I 各行之间相互正交，且为单位向量

weight_attr = paddle.ParamAttr(initializer=paddle.nn.initializer.Orthogonal())
linear = paddle.nn.Linear(15, 10, weight_attr=weight_attr)
# linear.weight: X' * X = I  各列之间相互正交，且为单位向量

weight_attr = paddle.ParamAttr(initializer=paddle.nn.initializer.Orthogonal())
linear = paddle.nn.Linear(10, 10, weight_attr=weight_attr)
# 各行、各列之间相互正交，且均为单位向量

```

paddle.nn.initializer.calculate_gain
---

infoflow 2021-11-18 21-42-23

```
import paddle

gain = paddle.nn.initializer.calculate_gain('tanh')
initializer = paddle.nn.intializer.Orthogonal(gain)
```

### CODE REVIEW

File: python/paddle/fluid/initializer.py [hunk 1]
```diff
@@ -1033,6 +1034,49 @@ def _global_bias_initializer():
...
+            import paddle
+            gain = paddle.nn.initializer.calculate_gain('tanh')
+            initializer = paddle.nn.initializer.Orthogonal(gain)
+
+    """
+    if param is None:
+        param = 0.01
+    else:
+        assert isinstance(param, (bool, int, float))
+    recommended_gain = {
```
Comment (jeff41404) [inline python/paddle/fluid/initializer.py:1062]: shall we support calculate gain of selu?

File: python/paddle/fluid/tests/unittests/test_initializer.py [hunk 2]
```diff
@@ -41,6 +41,13 @@ def output_hist(out):
 
 
 class TestConstantInitializer(unittest.TestCase):
+    def test_calculate_gain(self):
```
Comment (jeff41404) [inline python/paddle/fluid/tests/unittests/test_initializer.py:45]: shall we add test case of relu and selu?

File: python/paddle/fluid/initializer.py (hunk 1 above)
Comment (zhwesky2010) [reply python/paddle/fluid/initializer.py:1062]: done

File: python/paddle/fluid/tests/unittests/test_initializer.py (hunk 2 above)
Comment (zhwesky2010) [reply python/paddle/fluid/tests/unittests/test_initializer.py:45]: done, thx

**Review by jeff41404 — DISMISSED** [review]
lgtm

**Review by XiaoguangHu01 — DISMISSED** [review]
LG API

File: python/paddle/fluid/initializer.py [hunk 3]
```diff
@@ -1033,6 +1034,50 @@ def _global_bias_initializer():
     return _global_bias_initializer_
 
 
+def calculate_gain(nonlinearity, param=None):
+    """
+    Get the recommended gain value of some nonlinearity function.
+
+    Args:
+        nonlinearity: nonlinearity function.
+        param: optional parameter for somme nonlinearity function
```
Comment (TCChenlong) [inline python/paddle/fluid/initializer.py]: param(type, optional);
default is None, means xxx

File: python/paddle/fluid/initializer.py [hunk 4]
```diff
@@ -1033,6 +1034,50 @@ def _global_bias_initializer():
...
+
+    Returns:
+        The recommended gain value for nonlinearity function.
+
+    Examples:
+        .. code-block:: python
+
+            import paddle
+            gain = paddle.nn.initializer.calculate_gain('tanh')
+            initializer = paddle.nn.initializer.Orthogonal(gain)
```
Comment (TCChenlong) [inline python/paddle/fluid/initializer.py]: 最好能以注释给一下运行的结果

File: python/paddle/fluid/initializer.py (hunk 3 above)
Comment (zhwesky2010) [reply python/paddle/fluid/initializer.py]: done, thx

File: python/paddle/fluid/initializer.py (hunk 4 above)
Comment (zhwesky2010) [reply python/paddle/fluid/initializer.py]: done

**Review by TCChenlong — DISMISSED** [review]
LGTM for API Docs

**Review by TCChenlong — APPROVED** [review]
LGTM for API docs

**Review by XiaoguangHu01 — APPROVED** [review]
LG API

**Pull request merged**